src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,238 @@
-        return false;
+        Node<T> nodeToRemove = getNodeByValue(object);
+        if (nodeToRemove == null) {
+            return false;
+        }
+        unlink(nodeToRemove);
+        return true;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(T value, Node<T> next, Node<T> prev) {
+            this.item = value;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    private void linkBefore(T value, Node<T> next) {
+        Node<T> previousNode = next.prev;
+        Node<T> newNode = new Node<>(value, next, previousNode);
+        next.prev = newNode;
+        if (previousNode == null) {
+            head = newNode;
+        } else {
+            previousNode.next = newNode;
+        }
+        size++;
+    }
+
+    private void linkNewHead(T value) {
+        Node<T> first = head;
+        Node<T> newNode = new Node<>(value, head, null);
+        head = newNode;
+        if (first == null) {
+            tail = newNode;
+        } else {
+            first.prev = newNode;
+        }
+        size++;
+    }
+
+    private void linkNewTail(T value) {
+        Node<T> newNode = new Node<>(value, null, tail);
+        tail.next = newNode;
+        tail = newNode;
+        size++;
+    }
+
+    private T unlink(Node<T> nodeToUnlink) {
+        final T removeValue = nodeToUnlink.item;
+        Node<T> nextNode = nodeToUnlink.next;
+        Node<T> previous = nodeToUnlink.prev;
+        if (previous == null) {
+            head = nextNode;
+        } else {
+            previous.next = nextNode;
+            nodeToUnlink.prev = null;
+        }
+
+        if (nextNode == null) {
+            tail = previous;
+        } else {
+            nextNode.prev = previous;
+            nodeToUnlink.next = null;
+        }
+        nodeToUnlink.item = null;
+        size--;
+        return removeValue;
+    }
+
+    private T unlinkHead(Node<T> nodeToUnlink) {
+        final T removeItem = nodeToUnlink.item;
+        Node<T> nextNode = nodeToUnlink.next;
+        nodeToUnlink.item = null;
+        nodeToUnlink.next = null;
+        head = nextNode;
+        if (nextNode == null) {
+            tail = null;
+        } else {
+            nextNode.prev = null;
+        }
+        size--;
+        return removeItem;
+    }
+
+    private T unlinkTail(Node<T> nodeToUnlink) {
+        final T removeItem = nodeToUnlink.item;
+        Node<T> prev = nodeToUnlink.prev;
+        nodeToUnlink.item = null;
+        nodeToUnlink.prev = null;
+        tail = prev;
+        if (prev == null) {
+            head = null;
+        } else {
+            prev.next = null;
+        }
+        size--;
+        return removeItem;
+    }
```

Comment:
It might be done in way less code.

    private void unlink(Node<T> node) {
        if (node.equals(head)) {
            head = head.next;
        } else if (node.equals(tail)) {
            tail = tail.prev;
        } else {
            node.prev.next = node.next;
            node.next.prev = node.prev;
        }
    }
    It won't work in your code, but for example

Reply:
Simplified, but l will miss those methods.